vector/src/main/java/im/vector/app/core/date/DefaultDateFormatterProvider.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.date
+
+import android.content.Context
+import android.text.format.DateFormat
+import im.vector.app.core.resources.LocaleProvider
+import org.threeten.bp.format.DateTimeFormatter
+import javax.inject.Inject
+
+class DefaultDateFormatterProvider @Inject constructor(private val context: Context,
+                                                       private val localeProvider: LocaleProvider)
+    : DateFormatterProvider {
+
+    override val dateWithMonthFormatter: DateTimeFormatter by lazy {
```

Comment:
Should be nice to add example of formatted date in the Javadoc of all this methods, or maybe document the enum `DateFormatKind` please

Reply:
Yes